### DISCUSSION TITLE
How to use dpdata to build dataset with lots of Gaussian logs?

### DISCUSSION THREAD

**Discussion by PenguinW1105** [deepmodeling/dpdata#623] [Q&A]
Now, I have thousands of Gaussian logs files. How can I turn them into dataset once? Only can I turn only one of them into npy or raw. I try to use a python script but failed.

**Comment by phys-chem** [comment]
I think you can try a loop to read log files. You can append each return to a list, and then convert the list object once. Have you converted one log file successfully?

**Comment by IJully** [comment]
Have you succed？I want to do this too.

**Comment by sunzichen2004** [comment]
maybe using labeledsystem.append(system) could help. here is the code writen by Chatgpt tested by me.

directory = "./"
spout_files = [os.path.join(directory, f) for f in os.listdir(directory) if f.endswith(".out")]
labeled_system = dpdata.LabeledSystem(file_name=spout_files[0], fmt="orca/spout")
for spout_file in spout_files[1:]:
    system = dpdata.LabeledSystem(file_name=spout_file, fmt="orca/spout")
    labeled_system.append(system)
